docs/DID Communications Research and Specification/Interactions and Messages.md

```diff
@@ -0,0 +1,315 @@
+        "authentication": [{
+            "id": "did:iota:86b7t9786tb9JHFGJKHG8796UIZGUk87guzgUZIuggez#keys-1",
+            "type": "Ed25519VerificationKey2020",
+            "controller": "did:iota:86b7t9786tb9JHFGJKHG8796UIZGUk87guzgUZIuggez",
+            "publicKeyMultibase": "zH3C2AVvLMv6gmMNam3uVAjZpfkcJCwDwnZn6z3wXmqPV"
+        }]
+    }
+}
+```
+
+---
+## Authentication
+
+Proving control over an identifier.
+
+The authentication flow consists of a simple request-response message exchange, where the contents of the response must match those of the request. Because all messages are signed and authenticated, the response functions as proof of control by nature of being correctly signed by the keys listed in the DID Document of the issuer. Because of this, in scenarios where a more complex functionality (e.g. Credential Verification) is needed, an additional authentication flow is not necessary.
+
+### Roles
+- <u>**Verifier**</u>: Agent who requests and verifies the authenticity of the <u>authenticator</u>
+- <u>**Authenticator**</u>: Agent who proves control over their identifier
+
+### Messages
+
+#### authenticationRequest
+The <u>verifier</u> sends the `authenticationRequest` to the authentication service endpoint of the <u>authenticator</u>, specifying a `callbackURL` for the `authenticationResponse` to be posted to. The `thread` is used as a challenge and also serves to identity the `authenticationRequest`. The whole request is to be signed by the <u>authenticator</u>. 
+
+###### Layout
+
+```JSON
+authenticationRequest: {
+    "callbackURL",
+    "thread",
+    "challenge",
+    "id", // OPTIONAL!
+    "timing" // OPTIONAL!
+}
+```
+
+###### Example(s)
+
+```JSON
+{
+    "callbackURL": "https://www.aliceswonderland.com/auth",
+    "thread": "69-420-1337",
+    "challenge": "please sign this",
+    "id": "did:iota:86b7t9786tb9JHFGJKHG8796UIZGUk87guzgUZIuggez",
+    "timing": {
+        "out_time": "1991-04-20T13:37:11Z",
+        "expires_time": "2069-04-20T13:37:02Z",
+        "wait_until_time": "2069-04-20T13:37:00Z"
+    }
+}
+```
+
+#### authenticationResponse
+The <u>authenticator</u> answers with an `authenticationResponse`, providing a `signature` of the whole `authenticationRequest` JSON - the complete original `authenticationRequest`.
+
+###### Layout
+
+```JSON
+authenticationResponse: {
+    "thread",
+    "signature"
+}
+```
+
+###### Example(s)
+
+```JSON
+{
+    "thread": "69-420-1337",
+    "signature": {
+        "type": "JcsEd25519Signature2020",
+        "verificationMethod": "#authentication",
+        "signatureValue": "5Hw1JWv4a6hZH5obtAshbbKZQAJK6h8YbEwZvdxgWCXSL81fvRYoMCjt22vaBtZewgGq641dqR31C27YhDusoo4N"
+   }
+}
+```
+
+The `signature` provided here must correspond to the `#authentication` public key provided in the DID Document of the identity that the <u>verifier</u> has received earlier. If that is the case, the identifier is authenticated successfully.
```

Comment:
I actually don't want to reuse the #authentication key outside of transaction signing. We might want to consider some renaming on AuthChain and Authentication key as the term fits better here. Why would we enforce any type of key here?